intuita-inc/cal.com-demo
# Chore/auth pages group

## What does this PR do?

This PR migrates the /auth page group to use the app directory (which runs under the App Router).
Migrated pages exist under `/future/auth` and are accessible at all times.

## Type of change


## How should this be tested?

Pages performance can be accessed via their individual URLs and various metrics could be validated to be improved compared to the previous implementation.

If relevant env variables are set to 1 and AB_TEST_BUCKET_PROBABILITY is set to 100, then the future signup page should be rendered in place of the /signup page WITHOUT a visible redirect. We can see it by inspecting the existence of the x-calcom-future-routes-enabled cookie.

## Mandatory Tasks

## Checklist

- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
